presto-cassandra/src/main/java/com/facebook/presto/cassandra/CassandraMetadata.java

```diff
@@ -198,10 +204,88 @@ public ColumnMetadata getColumnMetadata(ConnectorSession session, ConnectorTable
     {
         CassandraTableHandle handle = checkType(table, CassandraTableHandle.class, "table");
         CassandraPartitionResult result = partitionManager.getPartitions(handle, constraint.getSummary());
-        ConnectorTableLayout layout = getTableLayout(session, new CassandraTableLayoutHandle(handle, result.getPartitions()));
+        ConnectorTableLayout layout = getTableLayout(session, new CassandraTableLayoutHandle(handle, result.getPartitions(), buildClusteringPredicate(schemaProvider.getTable(handle), constraint)));
```

Comment:
I think you don't need to build clustering key predicate when it is unpartitioned full-scan.